docs/architecture/distributed-cloud-native-apps-containers/api-gateways/gateway-patterns.md

```diff
@@ -0,0 +1,164 @@
+
+![Backend for Frontend Pattern](./media/backend-for-frontend-pattern.png)
+
+**Figure 13-3**. Backend for frontend pattern
+
+Note in the previous figure how incoming traffic is sent to a specific API gateway - based upon client type: web, mobile, or desktop app. This approach makes sense as the capabilities of each device differ significantly across form factor, performance, and display limitations. Typically mobile applications expose less functionality than a browser or desktop applications. Each gateway can be optimized to match the capabilities and functionality of the corresponding device.
+
+The backend for frontend pattern can be extended to be platform-specific. For example, an iOS and Android gateway could be created to expose platform-specific functionality. Or if you have specific third parties that need to access your services, you could create a gateway specifically for each of them.
+
+## Simple Gateways
+
+To start, you could build your own API Gateway service. A quick search of GitHub will provide many examples.
+
+For simple .NET cloud-native applications, you might consider the [Ocelot Gateway](https://github.com/ThreeMammals/Ocelot). Open source and created for .NET microservices, it's lightweight, fast, scalable. Like any API Gateway, its primary functionality is to forward incoming HTTP requests to downstream services. Additionally, it supports a wide variety of capabilities that are configurable in a .NET middleware pipeline.
+
+[YARP](https://github.com/microsoft/reverse-proxy) is another open source reverse proxy led by a group of Microsoft product teams. Downloadable as a NuGet package, YARP plugs into the ASP.NET framework as middleware and is highly customizable. You'll find YARP [well-documented](https://microsoft.github.io/reverse-proxy/articles/getting-started.html) with various usage examples. You'll see some examples of how to use YARP in the next section.
+
+For enterprise cloud-native applications, there are several managed Azure services that can help jump-start your efforts.
+
+## Azure Application Gateway
+
+For simple gateway requirements, you may consider [Azure Application Gateway](/azure/application-gateway/overview). Available as an Azure [PaaS service](https://azure.microsoft.com/overview/what-is-paas/), it includes basic gateway features such as URL routing, SSL termination, and a Web Application Firewall. The service supports [Layer-7 load balancing](https://www.nginx.com/resources/glossary/layer-7-load-balancing/) capabilities. With Layer 7, you can route requests based on the actual content of an HTTP message, not just low-level TCP network packets.
```

Comment:
Azure Application Gateway link is not working

Reply:
this will direct to https://learn.microsoft.com/en-us/azure/application-gateway/overview when live which should work